genrl/core/buffers.py

```diff
@@ -257,11 +251,84 @@ def update_priorities(self, batch_indices: Tuple, batch_priorities: Tuple) -> No
     def __len__(self) -> int:
         """
         Gives number of experiences in buffer currently
-
         :returns: Length of replay memory
         """
         return len(self.buffer)
 
     @property
     def pos(self):
         return len(self.buffer)
+
+
+class ReverbReplayBuffer:
+    def __init__(
+        self,
+        size,
+        batch_size,
+        obs_shape,
+        action_shape,
+        discrete=True,
+        reward_shape=(1,),
+        done_shape=(1,),
+        n_envs=1,
```

Comment:
Instead of passing all these, maybe it would be easier to get them all from the agent as `self.env`